cachi2/core/package_managers/rubygems/parser.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from dataclasses import dataclass
+from typing import Optional, Union
+
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+
+GEMFILE_LOCK = "Gemfile.lock"
+
+
+@dataclass(init=False)
```

Comment:
Do we want to keep `init=False`? If we do, we wouldn't be able to do:

```
GemDependency(name="abc", version="1.0")
```

Reply:
That's because `source` attribute is not optional.

Reply:
Dumb me, this is just the base class, I see the `@dataclass` (without init=False) keywords on the derived classes now.